Friends functionality for the back end

chobs95 opened the pull request.

chobs95 committed:
First stage of implementing the friends system into the backend

chobs95 committed:
second stage of friends back end. Should all work now.

QS-Coding committed:
Merge branch 'main' into friends-pages

QS-Coding reviewed (approved):
Looks good. I made sure the merge doesn't have any conflicts when merging with main

chobs95 commented:
Added the back end friends functionality. With this I have added functions to controllers/users.js: 
getAllUsers - fetches all users
getUser - gets a specific user
getUserFriends - gets friends of a specified user
addRemove friend - adds or removes a friends connection (updates both user and friend simultaneously)

Then within the user routes I updated this to match.
